storage/innobase/include/row0mysql.h

```diff
@@ -681,6 +681,12 @@ struct mysql_row_templ_t {
 					Innobase record in the current index;
 					not defined if template_type is
 					ROW_MYSQL_WHOLE_ROW */
+	bool	rec_field_is_prefix;	/* is this field in a prefix index? */
```

Comment:
Right now, rec_prefix_field_no == rec_field_no != ULINT_UNDEFINED for full fields as returned by dict_index_get_nth_col_pos. Checking whether it's possible to make ti return rec_prefix_field_no == ULINT_UNDEFINED for full fields.